src/hazelcore/semantics/CursorInfo.re

```diff
@@ -121,6 +129,10 @@ type t = {
   subblock_start: option(int),
 };
 
+type deferrable_t =
```

Comment:
Let's (1) move this definition to just before the first function that uses it and (2) include a comment describing what is going on here.